common/testing/testvars/test_vars.go

```diff
@@ -104,38 +90,74 @@ func (tv *TestVars) clone() *TestVars {
 
 	return tv2
 }
+func (tv *TestVars) cloneSetVal(key string, val any) *TestVars {
+	tv2 := tv.clone()
+	tv2.kv.Store(key, val)
+	return tv2
+}
 
-func (tv *TestVars) cloneSet(typ string, key []string, val any) *TestVars {
-	newTv := tv.clone()
-	newTv.set(typ, key, val)
-	return newTv
+func (tv *TestVars) cloneAppendString(key string, initialValGen func(key string) string, suffix string) *TestVars {
+	tv2 := tv.clone()
+
+	v, isLoaded := tv.kv.Load(key)
+	if !isLoaded {
+		v = initialValGen(key)
+	}
+
+	vString, vIsString := v.(string)
+	if !vIsString {
+		vStringer, vIsStringer := v.(fmt.Stringer)
+		if !vIsStringer {
+			panic(fmt.Sprintf("value of key %s is of type %T but must be of type %T or implement fmt.Stringer", key, v, ""))
+		}
+		vString = vStringer.String()
+	}
+	tv2.kv.Store(key, vString+"_"+suffix)
+
+	return tv2
 }
 
-// ----------- Methods for every "type" ------------
-// TODO: add more as you need them.
+// ----------- Methods for every entity ------------
+// Add more as you need them following the pattern below.
+// Replace "Entity" with the name of the entity, i.e., UpdateID, ActivityType, etc.
+// Add only the necessary methods (in most cases only getter).
+/*
+func (tv *TestVars) Entity() string {
+	return getOrCreate(tv, "entity", tv.uniqueString)
+}
+func (tv *TestVars) WithEntity(entity string) *TestVars {
```

Comment:
usually "With" semantic means you are operating within the same object, rather then creating new object. It is kind of part of the constructor ("chaining" pattern). 
Code should look like
tv:=testvars.New(testvars.WithWorkflowId(), testvars.WithWorkflowId(), ...)

Reply:
those are main comments, rest are oservations.